autopts/pybtp/btp/ccp.py

```diff
@@ -0,0 +1,255 @@
+#
+# auto-pts - The Bluetooth PTS Automation Framework
+#
+# Copyright (c) 2023, Oticon.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms and conditions of the GNU General Public License,
+# version 2, as published by the Free Software Foundation.
+#
+# This program is distributed in the hope it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+
+"""Wrapper around btp messages. The functions are added as needed."""
+import binascii
+import logging
+import struct
+from enum import IntEnum, IntFlag
+from threading import Timer, Event
+from time import sleep
+
+from autopts.pybtp import defs
+from autopts.pybtp.btp.btp import CONTROLLER_INDEX, get_iut_method as get_iut, btp_hdr_check
+
+
+CCP = {
+    'read_supported_cmds': ( defs.BTP_SERVICE_ID_CCP,
+                             defs.CCP_READ_SUPPORTED_COMMANDS,
+                             CONTROLLER_INDEX),
+    'register_callbacks':  ( defs.BTP_SERVICE_ID_CCP,
+                             defs.CCP_REGISTER_CALLBACKS,
+                             CONTROLLER_INDEX),
+    'discover_tbs':        ( defs.BTP_SERVICE_ID_CCP,
+                             defs.CCP_DISCOVER_TBS,
+                             CONTROLLER_INDEX, ''),
+    'accept_call':         ( defs.BTP_SERVICE_ID_CCP,
+                             defs.CCP_ACCEPT_CALL,
+                             CONTROLLER_INDEX),
+    'terminate_call':      ( defs.BTP_SERVICE_ID_CCP,
+                             defs.CCP_TERMINATE_CALL,
+                             CONTROLLER_INDEX),
+    'originate_call':      ( defs.BTP_SERVICE_ID_CCP,
+                             defs.CCP_ORIGINATE_CALL,
+                             CONTROLLER_INDEX),
+    'read_call_state':     ( defs.BTP_SERVICE_ID_CCP,
+                             defs.CCP_READ_CALL_STATE,
+                             CONTROLLER_INDEX)
+}
+
+events = {
+    'Discovered': { 'count': 0, 'status': 0, 'tbs_count': 0, 'gtbs': False },
+    'Call_State': { 'count': 0, 'status': 0, 'index': 0, 'call_count': 0, 'states': [] }
+}
```

Comment:
Maybe events should be implemented as a part of Stack class, so they could be accessed easily across layers
https://github.com/auto-pts/auto-pts/blob/263c5c33d279eef423a0d23d512f9bf01d3d1e86/autopts/ptsprojects/stack.py#L1353-L1369

Like it has been done for previous layers, e.g.:
https://github.com/auto-pts/auto-pts/blob/263c5c33d279eef423a0d23d512f9bf01d3d1e86/autopts/ptsprojects/stack.py#L704-L710

Reply:
Done.